src/Database/Dialect/TupleComparisonTranslatorTrait.php

```diff
@@ -79,7 +79,7 @@ protected function _transformTupleComparison(TupleComparison $expression, $query
 
         foreach ($value as $tuple) {
             $surrogate->orWhere(function ($exp) use ($fields, $tuple) {
-                foreach ($tuple as $i => $value) {
+                foreach (array_values($tuple) as $i => $value) {
                     $exp->add([$fields[$i] => $value]);
```

Comment:
If we check it it would be to throw an exception as it means something broke elsewhere. Here it was just assuming it was a zero-index array